---
int32 stream_mode